### ISSUE TITLE
 "Found built image, launching..." but the session is not launching

### ISSUE THREAD

**Issue by melaniewalsh** [jupyterhub/binderhub#1234]
### Bug description

This morning I shared my [Jupyter Book](https://melaniewalsh.github.io/Intro-Cultural-Analytics/welcome.html) project on social media -- all of the notebooks included in the book have Binder instances -- but Binder is not launching for the notebooks. The build log says "Found built image, launching..." but the instance never launches. Maybe this is happening because more people than usual are trying to launch Binder instances? Any suggestions for fixing this issue would be much appreciated.

### How to reproduce

1. Go to https://melaniewalsh.github.io/Intro-Cultural-Analytics/Data-Analysis/Pandas-Basics-Part1.html
2. Click on the Binder instance in the top right corner https://mybinder.org/v2/gh/melaniewalsh/Intro-Cultural-Analytics/master?urlpath=lab/tree/book/Data-Analysis/Pandas-Basics-Part1.ipynb
3. See "Found built image, launching..."

### Your personal set up

 - OS: MacOS 10
 

- <details><summary>Logs</summary>

```
"Found built image, launching..."
```

</details>

**Comment by melaniewalsh** [comment]
Just an update that it seems to be working again. For the future: does number of users impact Binder load time?

**Comment by betatim** [comment]
Hi 👋 and what a cool book/project!

We limit the number of simultaneous instances of one repository to 100. When a repo reaches that limit the build log will say so. This means that this is/was a different thing.

Usually the "Found image, launching ..." means that the image is being launched but for some reason isn't reporting back "hey I am alive now, send the user over!!". Most of the time this is because there is a problem with the entrypoint/command of the image. However if your image worked and then stopped working without you making changes, this is probably not it.

I'll take a look at the logs to see what I can see.

tl;dr: not sure, probably not your fault, probably related to how popular the repo got (in a short amount of time).

**Comment by melaniewalsh** [comment]
Thanks, Tim (@betatim)! 🎉 Binder is a huge part of what makes the book/project cool (in my humble opinion)!

This is very helpful to know re simultaneous instances and the "Found image, launching..." message. I'm sure that I can also make my project more streamlined for faster launch times. If you find anything in the logs with regard to that kind of improvement, I would love to know. I'm still learning, and my project is still growing!

**Comment by betatim** [comment]
I think you might have been a victim of an upgrade of a dependency we use to talk to the Kubernetes API. At least I can't find anything else that looks out of the ordinary.

To make launches faster there are two things you can investigate:
1. not reference `master`/`main` but a tag. This way not every typo fix triggers a rebuild. Instead you can periodically move the tag (say once a week or so)
2. investigate making your image smaller via something like https://github.com/binder-examples/minimal-dockerfile. Smaller images are faster to transfer over the network which is one of the main time sinks after building the image

(2) speeds up launches a bit but means you are going the manual route, so some of the "magic" of binder (actually repo2docker) is lost. However with the increased control you can shrink the image a lot. Maybe something to investigate once things are becoming stable and development of the book slows down a bit?

(If you are happy please close the issue to help keep things a bit tidy :D We should continue optimisation talk in the forum on discourse.jupyter.org as that is more discoverable and has more readers who have ideas)

**Comment by melaniewalsh** [comment]
Thanks so much! These are great tips for optimizing the Binder launch time, and it's good to know about the upgrade. I will take any further questions or discussion to the Discussion Forum! Cheers.
